### DISCUSSION TITLE
Notification triggered by server?

### DISCUSSION THREAD

**Discussion by krlmlr** [woodpecker-ci/woodpecker#4198] [Ask a Question] [answered]
Thanks for developing and maintaining Woodpecker!

We're using Woodpecker with Gitea. I wonder if there is a way to get a notification of any kind after a successful/failing/... run for any repo configured. A webhook would do fine, doesn't have to be e-mail. Can I perhaps somehow teach Gitea to forward Woodpecker's notifications?

https://github.com/woodpecker-ci/woodpecker/discussions/3623 suggests using a plugin for this, but I'm specifically looking for a way to configure the system, not the individual repositories.

My plan would be to use a tool like ntfy to forward those notifications as needed. I'm aware there's a ntfy plugin, but again I'm looking for a configuration of the system.

**Comment by qwerty287** [comment, marked as answer]
Had that idea as well, there's https://github.com/woodpecker-ci/woodpecker/issues/1070 open. So this is not really possible currently. You could use a config extension to append a webhook step/workflow to every pipeline

**Reply by krlmlr** [reply]
Thanks! What is a "config extension"? #783 still looks unresolved.

**Reply by qwerty287** [reply]
Yes, there will likely be some changes to this in next versions, but currently it works like this: https://woodpecker-ci.org/docs/administration/advanced/external-configuration-api

**Comment by anbraten** [comment]
This FR might also be sth you are looking for: https://github.com/go-gitea/gitea/issues/20749
